js/ui/codingManager.js

```diff
@@ -942,10 +1213,16 @@ const CodingManager = new Lang.Class({
         if (!_isCodingApp(window.get_flatpak_id()))
             return false;
 
-        this._sessions.push(new CodingSession({
+        let session = new CodingSession({
             app: actor,
-            button: new WindowTrackingButton({ window: actor.meta_window })
-        }));
+            button: new WindowTrackingButton({ window: actor.meta_window }),
+            install_monitor: new CodingInstallationMonitor({
+                flatpak_name: window.get_flatpak_id()
+            })
+        });
+        this._sessions.push(session);
+
+        this.emit('session-created', session);
```

Comment:
Instead of using this signal here, you can just add a method to the `CodingOSIntegration` object and call it; no other client either listens to this or uses `CodingOSIntegration`